pkg/devfile/parser/parse.go

```diff
@@ -914,3 +931,124 @@ func getKubernetesDefinitionFromUri(uri string, d devfileCtx.DevfileCtx) ([]byte
 	}
 	return data, nil
 }
+
+// Write Docker-Compose content to Temporary directory
+func WriteContentToTempDir(fs filesystem.Filesystem, content string) (string, error) {
+	tempDir, err := fs.TempDir("./", "temp")
+	if err != nil {
+		return "", err
+	}
+
+	filePath := path.Join(tempDir, "docker-compose.yaml")
+
+	file, err := fs.Create(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+
+	_, err = file.WriteString(content)
+	if err != nil {
+		return "", err
+	}
+
+	//file.Close()
+
+	return filePath, err
+}
+
+// Identify and iterate through each occurence of docker compose component and convert to k8s represenation by convertDockerComposeToK8s function
+func parseDockerCompose(devObj DevfileObj) error {
+	//parse Devfile
+	getDockercomposeOptions := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1.ComposeComponentType,
+		},
+	}
+	//parse for docker compose components
+	dockercomposeComponents, err := devObj.Data.GetComponents(getDockercomposeOptions)
+	if err != nil {
+		return err
+	}
+	//iterate through each docker compose component
+	for _, dockercomposeComp := range dockercomposeComponents {
```

Comment:
Consider breaking this up into multiple functions (e.g. a function for handling URIs), I'm finding it hard to parse with all the conditionals.